tests/utils/test.py

```diff
@@ -69,9 +69,14 @@ def parse_response_to_soup(response, selector=None, no_html_body=False, replace_
     for csp_nonce_script in soup.find_all("script", {"nonce": True}):
         csp_nonce_script["nonce"] = "NORMALIZED_CSP_NONCE"
     if replace_objects_pk_in_href:
-        for links in soup.find_all(attrs={"href": True}):
-            for obj in replace_objects_pk_in_href:
-                links.attrs["href"] = links.attrs["href"].replace(str(obj.pk), f"[PK of {type(obj).__name__}]")
+        if all(isinstance(item, tuple) for item in replace_objects_pk_in_href):
+            for links in soup.find_all(attrs={"href": True}):
+                for origin, replacement in replace_objects_pk_in_href:
+                    links.attrs["href"] = links.attrs["href"].replace(origin, replacement)
+        else:
+            for links in soup.find_all(attrs={"href": True}):
+                for obj in replace_objects_pk_in_href:
+                    links.attrs["href"] = links.attrs["href"].replace(str(obj.pk), f"[PK of {type(obj).__name__}]")
```

Comment:
Plutôt que de fonctionner en "tout tuple" ou "tout objet", j'autoriserais le mélange:

```suggestion
        for links in soup.find_all(attrs={"href": True}):
            for replace_to_do in replace_objects_pk_in_href:
                if isinstance(replace_to_do, tuple):
                    origin, replacement = replace_to_do
                else:
                    origin, replacement = str(replace_to_do.pk), f"[PK of {type(replace_to_do).__name__}]"
                links.attrs["href"] = links.attrs["href"].replace(origin, replacement)
```

(et au passage de je renommerais `replace_objects_pk_in_href` en `replace_in_href`)